src/Config.php

```diff
@@ -142,6 +144,24 @@ public function registerCustomFixers(iterable $fixers): ConfigInterface
         return $this;
     }
 
+    /**
+     * Registers custom rule sets to be used the same way as built-in rule sets.
+     *
+     * `$ruleSets` must follow `'@RuleName' => RuleSetDescriptionInterface::class` convention.
+     *
+     * @param array<string, class-string<RuleSetDescriptionInterface>> $ruleSets
+     *
+     * @todo Introduce it in ConfigInterface in 4.0
+     */
+    public function registerCustomRuleSets(array $ruleSets): ConfigInterface
+    {
+        foreach ($ruleSets as $name => $class) {
+            RuleSets::registerRuleSet($name, $class);
```

Comment:
Because `Config::registerCustomRuleSets()` is not a part of `ConfigInterface` at this point (would be a BC-break) and it only proxies call to `RuleSets::registerRuleSet()`. If anyone uses different `ConfigInterface` implementation, also should use `RuleSets::registerRuleSet()` call in ones' config.

How would you want to achieve this without some static access? Maybe I missed something.

Reply:
This explanation makes sense.

I wonder if anyone is using they own `ConfigInterface` implementation.

Reply:
For me it's perfect example of over-engineering and this interface shouldn't exist 😅. Why should any other implementation be allowed? 🤷‍♂️